src/gt4sd/frameworks/enzeptional/processing.py

```diff
@@ -242,16 +378,76 @@ def reconstruct_sequence_with_mutation_range(
         reconstructed sequence.
     """
     # create the mutated sequence, considering sorted intervals
-    mutated_range_offset = 0  # offset with respect to the mutated_sequence_range
-    mutated_sequence_offset = 0  # offset with respect to the full mutated sequence.
-    mutated_sequence = ""
+    mutated_sequence = list(sequence)
+    # split fragments by intervals
     for start, end in intervals:
-        mutated_sequence += sequence[mutated_sequence_offset:start]
-        chunk_length = end - start + 1
-        mutated_sequence += mutated_sequence_range[
-            mutated_range_offset : mutated_range_offset + chunk_length
-        ]
-        mutated_range_offset += chunk_length
-        mutated_sequence_offset = end + 1
-    mutated_sequence += sequence[end + 1 :]
-    return mutated_sequence
+        size_fragment = end - start
+        mutated_sequence[start:end] = list(mutated_sequence_range[:size_fragment])
+        mutated_sequence_range = mutated_sequence_range[size_fragment:]
+    return "".join(mutated_sequence)
+
+
+def selection(scores: List[Dict[str, Any]], k: int = 20) -> List[Any]:
```

Comment:
not a fan of having a function for a one liner, but it would make sense to have a selector class that allows for implementing various selectors and defaults to this one in the base.